### ISSUE TITLE
[Bug]: FCM credentials not persisted in config entry data

### ISSUE THREAD

**Issue by raven2k24** [bvis/aegis-hass#183]
### Describe the bug

I am saving the 4 FCM entries. I see them when i go on the settings again. Reloaded, restarted hass but...

custom_components.aegis_ajax.notification
FCM credentials not configured — push notifications disabled, real-time events 
(doorbell ring, arm/disarm, alarm) will not reach HA. Configure them in 
Settings → Devices & Services → Aegis for Ajax → Configure, or open the 
Repair card surfaced under Settings → Repairs.

Source: custom_components/aegis_ajax/notification.py, line 176
Level: WARNING
First occurred: 2026-05-23 21:05:53 (ora României)
Count: 1

### Steps to reproduce

version 1.5.1
the bug was present in 1.5.0 also

### Expected behavior

FCM should work

### Home Assistant version

2026.5

### Integration version

1.5.1

### Installation method

HACS

### App label (co-branded app)

Ajax

### FCM push notifications configured?

Yes

### Relevant logs

```text

```

**Comment by bvis** [comment]
Likely real, and the cause is subtle. When you re-open Options, HA pre-fills `Project ID` / `App ID` / `Sender ID` with the saved values (visible) but leaves `API Key` blank by design — it's a password field, and HA's frontend never displays a saved secret, even masked. So "I see them when I go on the settings again" only confirms 3 of 4 are persisted.

The likely sequence:

1. First open of Options → you typed all 4 values → submitted → saved correctly.
2. Re-opened Options to verify → the API Key field appeared blank (normal — HA hides it).
3. Clicked Submit again → the empty submission for the password field wiped the saved API Key (the integration treats an empty value as "clear this one" for that field, a side-effect of an earlier fix in [#138](https://github.com/bvis/aegis-hass/issues/138)).
4. After reload / restart → 3 of 4 keys remain in entry data → `notification.py` warns "FCM credentials not configured".

To confirm, please paste the `entry_data` section from the diagnostics download — the four FCM keys are already redacted in the file, but you'll see whether `fcm_api_key` appears as a key at all or is missing entirely:

1. **Settings → Devices & Services → Aegis for Ajax → ⋮ (three dots) → Download diagnostics**
2. Open the JSON, find the `entry_data` object, paste it here.

**Workaround in the meantime:** open Options once, re-enter the API Key (you'll need to re-paste from the APK extraction since HA doesn't expose the saved value), submit. Don't re-open Options afterwards unless you also re-type the API Key.

**Next beta** will treat an empty password submission as "leave alone" — only the explicit `Delete FCM credentials` toggle will wipe the saved value. The re-open-and-submit cycle will stop eating the API Key.

**Comment by bvis** [comment]
Quick follow-up: the fix landed in [`1.5.3-beta.5`](https://github.com/bvis/aegis-hass/releases/tag/v1.5.3-beta.5) (just released). An empty `FCM API Key` submission is now treated as "leave alone" — only the `Delete FCM credentials` toggle wipes the saved key. You should be able to:

1. Update to `1.5.3-beta.5` via HACS (Beta channel must be enabled — User Profile → Advanced → Show beta versions in HACS).
2. Open Options once, type all 4 FCM values, Submit. The integration reloads and FCM registration should go through.
3. After that, you can re-open Options to change anything else (poll interval, etc.) without re-typing the API Key — the saved value sticks now.

If you can still grab the `entry_data` from the diagnostics download before / after the update, that'll confirm whether the bug matches what we fixed (i.e. `fcm_api_key` was missing from the pre-update `entry_data` but the other three were present). The fix is independent of that confirmation, but it'll help close the loop on this issue.
